core/Pos/Core/Types.hs

```diff
@@ -248,11 +247,11 @@ type ProxySigHeavy a = ProxySignature EpochIndex a
 -- | Heavy delegation psk.
 type ProxySKHeavy = ProxySecretKey EpochIndex
 
--- | Some proxy secret key.
-type ProxySKEither = Either ProxySKLight ProxySKHeavy
-
--- | Map from issuers to heavy certs.
-type ProxySKHeavyMap = HashMap PublicKey ProxySKHeavy
+-- | Lightweight PSK or heavyweight PSK with real leader public key
```

Comment:
Yes, it's needed here. Before we were passing `(sk, Maybe (Either PskLight PskHeavy))`, where `sk` because psk has `issuerPk` inside. But now we can have slot leader `i` in the chain `i -> x -> d`, where we (sk) are `d`, and `x -> d` is heavy cert we pass, so we don't have any info about `i`. That's why it's added to this datatype.